cmd/sql-proxy-client/main.go

```diff
@@ -69,29 +66,31 @@ func realMain() error {
 		return errors.New("--token and --service-token/--service-token-name cannot be set at the same time")
 	}
 
-	if *orgName == "" || *dbName == "" || *branchName == "" {
-		return errors.New("--org, --database or --branch is not set")
-	}
-
 	var certSource proxy.CertSource
 	var err error
 
-	certSource, err = newRemoteCertSource(*token, *serviceToken, *serviceTokenName)
-	if err != nil {
-		return err
+	if *token != "" || (*serviceToken != "" && *serviceTokenName != "") {
+		certSource, err = newRemoteCertSource(*token, *serviceToken, *serviceTokenName)
+		if err != nil {
+			return err
+		}
 	}
 
-	if *caPath != "" && *clientCertPath != "" && *clientKeyPath != "" {
-		certSource, err = newLocalCertSource(*caPath, *clientCertPath, *clientKeyPath, *remoteAddr, *remotePort)
+	if *remoteHost != "" && *clientCertPath != "" && *clientKeyPath != "" {
+		certSource, err = newLocalCertSource(*clientCertPath, *clientKeyPath, *remoteHost, *remotePort)
 		if err != nil {
 			return err
 		}
 	}
 
+	if certSource == nil {
+		return fmt.Errorf("no configuration found, need either a token and org / datbase / branch parameters or separate specified certificate source and remote host")
+	}
+
 	p, err := proxy.NewClient(proxy.Options{
 		CertSource: certSource,
 		LocalAddr:  net.JoinHostPort(*host, *port),
-		RemoteAddr: *remoteAddr,
+		RemoteAddr: net.JoinHostPort(*remoteHost, strconv.Itoa(*remotePort)),
 		Instance:   fmt.Sprintf("%s/%s/%s", *orgName, *dbName, *branchName),
```

Comment:
It looks like it'll converge to `///` if you don't populate it. And then the cert will be cached for this name. I wonder if we should rename this if you select a local cert source?

Reply:
Good idea, fixed this up by using the subject on the certificate.